core/src/main/scala/io/finch/syntax/EndpointMapper.scala

```diff
@@ -8,7 +8,7 @@ class EndpointMapper[A](m: Method, e: Endpoint[A]) extends Endpoint[A] { self =>
   /**
    * Maps this endpoint to either `A => Output[B]` or `A => Future[Output[B]]`.
    */
-  final def apply(mapper: Mapper[A]): Endpoint[mapper.Out] = mapper(self)
+  final def apply[F](f: => F)(implicit mapper: Mapper[F, A]): Endpoint[mapper.Out] = mapper(f, self)
```

Comment:
The only concerning bit is this by-name value. I understand why we need it, but I wish we could do better than it. I'm worried about an overhead (for a wrapping function0) for a general case (map `A => B`).

Reply:
I think we can keep `ToTwitterFuture`. I actually like it.

Reply:
@vkostyukov are you still concerned with the overhead of the by-name parameter? If so maybe a few of us could run the benchmarks against this branch and master to get an idea of how much it adds?

Reply:
Yeah. I think it's a reasonable trade-off to make: better user experience (fewer highlighting erros) > +16 byte increase in allocations per endpoint. But yeah - running some benchmarks would be a good idea.

Reply:
I think we can add a new benchmark under `benchmarks` project.